### PR TITLE
[RDY] vim-patch:8.0.0100

### PR DESCRIPTION
**Pull request by ckelsel**
Problem:    Options that are a file name may contain non-filename characters.
Solution:   Check for more invalid characters.

https://github.com/vim/vim/commit/319afe3804741db5a6c188bd69535fa7ed044c62

### CODE REVIEW

**Comment by justinmk** [comment]
Doesn't seem RDY. We are missing the change. Compare our current code:

```
  else if ((options[opt_idx].flags & P_NFNAME)                                                                                                              
           && vim_strpbrk(*varp, (char_u *)"/\\*?[|<>") != NULL) {
```

**Comment by ckelsel** [comment]
@justinmk I am sorry,  missing to add and commit changed files. Now It's Fixed.

File: src/nvim/option.c [hunk 1]
```diff
@@ -2457,12 +2457,11 @@ did_set_string_option (
...
-  /* Check for a "normal" file name in some options.  Disallow a path
-   * separator (slash and/or backslash), wildcards and characters that are
-   * often illegal in a file name. */
-  else if ((options[opt_idx].flags & P_NFNAME)
-           && vim_strpbrk(*varp, (char_u *)"/\\*?[|<>") != NULL) {
+  } else if ((options[opt_idx].flags & P_NFNAME)
+             && vim_strpbrk(*varp, (char_u *)"/\\*?[|;&<>\r\n") != NULL) {
+    // Check for a "normal" file name in some options.  Disallow a path
+    // separator (slash and/or backslash), wildcards and characters that are
+    // often illegal in a file name.
```
Comment (justinmk) [inline src/nvim/option.c]: Just for future: normally this would stay about the `} else if ...` line.

Comment (ckelsel) [inline src/nvim/option.c]:
```
// Check for a "normal" file name in some options.  Disallow a path
// separator (slash and/or backslash), wildcards and characters that are
// often illegal in a file name.
} else if ((options[opt_idx].flags & P_NFNAME)
```

Ok, I see.

**Pull request merged**